src/main/java/edu/harvard/iq/dataverse/api/auth/BearerTokenAuthMechanism.java

```diff
@@ -0,0 +1,116 @@
+package edu.harvard.iq.dataverse.api.auth;
+
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.authorization.UserRecordIdentifier;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.OAuth2Exception;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc.OIDCAuthProvider;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.authorization.users.User;
+import edu.harvard.iq.dataverse.settings.FeatureFlags;
+
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.HttpHeaders;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class BearerTokenAuthMechanism implements AuthMechanism {
+    private static final String BEARER_AUTH_SCHEME = "Bearer";
+    public static final String UNAUTHORIZED_BEARER_TOKEN = "Unauthorized bearer token";
+    public static final String INVALID_BEARER_TOKEN = "Could not parse bearer token";
+    public static final String BEARER_TOKEN_DETECTED_NO_OIDC_PROVIDER_CONFIGURED = "Bearer token detected, no OIDC provider configured";
+
+    @Inject
+    protected AuthenticationServiceBean authSvc;
+    @Inject
+    protected UserServiceBean userSvc;
+    private static final Logger logger = Logger.getLogger(BearerTokenAuthMechanism.class.getCanonicalName());
+    @Override
+    public User findUserFromRequest(ContainerRequestContext containerRequestContext) throws WrappedAuthErrorResponse {
+        if (FeatureFlags.API_BEARER_AUTH.enabled()) {
+            Optional<String> bearerToken = getRequestApiKey(containerRequestContext);
+            // No Bearer Token present, hence no user can be authenticated
+            if (!bearerToken.isPresent()) {
+                return null;
+            }
+            //validate and verify provided Bearer Token, and retrieve UserRecordIdentifier
```

Comment:
@pdurbin @GPortas @kcondon @scolapasta should we address this now before merging or is it fine to let that go in as is? At least until now this is an experimental feature with a feature flag, so maybe OK to merge without this caching.

```suggestion
            // Validate and verify provided Bearer Token, and retrieve UserRecordIdentifier
            // TODO: Get the identifier from an invalidating cache to avoid lookup bursts of the same token. Tokens in the cache should be removed after some (configurable) time.
```